scripts/install-sealedsecrets.sh

```diff
@@ -1,13 +1,11 @@
 #!/bin/bash
 
-GOOS=$(go env GOOS)
-GOARCH=$(go env GOARCH)
-
 release=$(curl -sI https://github.com/bitnami-labs/sealed-secrets/releases/latest | grep Location | awk -F"/" '{ printf "%s", $NF }' | tr -d '\r')
 #release=$(curl --silent "https://api.github.com/repos/bitnami-labs/sealed-secrets/releases/latest" | sed -n 's/.*"tag_name": *"\([^"]*\)".*/\1/p')
 
 echo "SealedSecrets release: $release"
 
-kubectl create -f https://github.com/bitnami-labs/sealed-secrets/releases/download/$release/sealedsecret-crd.yaml
+helm del --purge ofc-bootstrap
+kubectl delete customresourcedefinition sealedsecrets.bitnami.com
```

Comment:
Same comment

Reply:
If I move it there I'm having
```
2019/01/28 19:07:52 Error: a release named ofc-bootstrap already exists.
Run: helm ls --all ofc-bootstrap; to check the status of the release
Or run: helm del --purge ofc-bootstrap; to delete it
```
That's why I've put it right before. Not sure if it's some bug with helm. Works the same however I name it.

Reply:
@stefanprodan any thoughts why that would happen? (This really shouldn't need to be here.)

Reply:
Curious, if you run the delete, what does `helm ls --all` say after?

Reply:
Can it be about the CRD being left over? `kubectl get crd`?

Reply:
The CRD does not get deleted and this is intentional, you don't want to lose all your secrets if some accidentally deletes the sealedsecrets release.

Reply:
I fond very confusing to name the sealedsecrets release `ofc-bootstrap` since sealedsecrets runs under a Cluster Role level. I would call it `ofc-sealedsecrets`

Reply:
SGMT. @ivanayov can you alter the release from `ofc-bootstrap` to `ofc-sealedsecrets` as per Stefan's comment?

Let's add the removal of the CRD to the reset script, so that it is only run there deliberately and removal the reset code from the installation step. (it definitely doesn't belong there)

Reply:
@alexellis Moving to the reset script doesn't solve the problem. These are more a workaround than part of the cluster reset.

After reset with these delete commands, I do
```
$ ./scripts/create-tiller-sa.sh
$ ./scripts/create-tiller.sh 
$ ./scripts/install-sealedsecrets.sh 
```
and it fails with
```
SealedSecrets release: v0.7.0
Error: release ofc-sealedsecrets failed: clusterroles.rbac.authorization.k8s.io "secrets-unsealer" already exists
```

CRDs after `./scripts/reset.sh && ./scripts/create-tiller-sa.sh && ./scripts/create-tiller.sh`:
```
$ kubectl get crd
NAME                                             CREATED AT
ciliumendpoints.cilium.io                        2019-02-07T10:06:49Z
ciliumnetworkpolicies.cilium.io                  2019-02-07T10:06:49Z
csidrivers.csi.storage.k8s.io                    2019-02-07T10:06:34Z
csinodeinfos.csi.storage.k8s.io                  2019-02-07T10:06:26Z
volumesnapshotclasses.snapshot.storage.k8s.io    2019-02-07T10:06:26Z
volumesnapshotcontents.snapshot.storage.k8s.io   2019-02-07T10:06:26Z
volumesnapshots.snapshot.storage.k8s.io          2019-02-07T10:06:26Z
```

and after `./scripts/install-sealedsecrets.sh `:
```
$ kubectl get crd
NAME                                             CREATED AT
ciliumendpoints.cilium.io                        2019-02-07T10:06:49Z
ciliumnetworkpolicies.cilium.io                  2019-02-07T10:06:49Z
csidrivers.csi.storage.k8s.io                    2019-02-07T10:06:34Z
csinodeinfos.csi.storage.k8s.io                  2019-02-07T10:06:26Z
sealedsecrets.bitnami.com                        2019-02-07T13:05:54Z
volumesnapshotclasses.snapshot.storage.k8s.io    2019-02-07T10:06:26Z
volumesnapshotcontents.snapshot.storage.k8s.io   2019-02-07T10:06:26Z
volumesnapshots.snapshot.storage.k8s.io          2019-02-07T10:06:26Z
```
i.e. `sealedsecrets.bitnami.com` is being deleted and not causing the problem.

Somehow running `helm install --namespace kube-system --name ofc-sealedsecrets stable/sealed-secrets` is causing the role to be created twice:

```
$ ./scripts/create-tiller.sh 
$HELM_HOME has been configured at /Users/iyovcheva/.helm.

Tiller (the Helm server-side component) has been upgraded to the current version.
Happy Helming!
```
```
$ helm ls --all
Error: could not find a ready tiller pod
```
```
$ helm install --namespace kube-system --name ofc-sealedsecrets stable/sealed-secrets
Error: a release named ofc-sealedsecrets already exists.
Run: helm ls --all ofc-sealedsecrets; to check the status of the release
Or run: helm del --purge ofc-sealedsecrets; to delete it
```
```
$  helm ls --all ofc-sealedsecrets
NAME             	REVISION	UPDATED                 	STATUS	CHART               	NAMESPACE  
ofc-sealedsecrets	1       	Thu Feb  7 15:05:54 2019	FAILED	sealed-secrets-1.0.1	kube-system
```

Reply:
Thank you for looking into this.

If the only thing creating an error is the cluster role, do we just need to remove that in the reset script?

Alex

Reply:
Would it help putting sealed-secrets in its own namespace?

Reply:
I get this on `kind`:

```
[CustomResourceDefinition] sealedsecrets.bitnami.com
```

giving:

```
helm install --namespace kube-system --name ofc-bootstrap stable/sealed-secrets
Error: release ofc-bootstrap failed: customresourcedefinitions.apiextensions.k8s.io "sealedsecrets.bitnami.com" already exists
```

Then:

```
 kubectl delete crd/sealedsecrets.bitnami.com
customresourcedefinition.apiextensions.k8s.io "sealedsecrets.bitnami.com" deleted
```

That then lets me install again:

```
helm install --namespace kube-system --name ofc-bootstrap stable/sealed-secrets
NAME:   ofc-bootstrap
LAST DEPLOYED: Thu Feb  7 14:33:32 2019
NAMESPACE: kube-system
STATUS: DEPLOYED
```